src/libs/actions/Search.ts

```diff
@@ -94,12 +94,34 @@ function createTransactionThread(hash: number, transactionID: string, reportID:
 function holdMoneyRequestOnSearch(hash: number, transactionIDList: string[], comment: string) {
     const {optimisticData, finallyData} = getOnyxLoadingData(hash);
 
+    optimisticData.push({
```

Comment:
I think we should probably also: 

1. Add an explanatory comment here
2. Only do these optimistic updates for the issue's case, e.g. via a new param `isSingleTransaction`
3. Optimistically add the comment to IOU's report (this can probably be copied from `HoldRequest` as this is what currently happens on staging when (un/)holding whilst offline)
![image](https://github.com/user-attachments/assets/ba83580e-3227-4d0c-9988-ed6a1d2a5e1a)

Reply:
Yeah it doesn't seem useful if it gets duplicated. Let's revert for now whilst we decide on the approach.